crates/api/src/proposer/api.rs

```diff
@@ -469,6 +469,141 @@ where
         }
     }
 
+    /// Retrieves the best bid header (with inclusion proof) for the specified slot, parent hash,
+    /// and public key.
+    ///
+    /// This function accepts a slot number, parent hash and public_key.
+    /// 1. Validates that the request's slot is not older than the head slot.
+    /// 2. Validates the request timestamp to ensure it's not too late.
+    /// 3. Fetches the best bid for the given parameters from the auctioneer.
+    /// 4. Fetches the inclusion proof for the best bid.
+    ///
+    /// The function returns a JSON response containing the best bid and inclusion proofs if found.
+    ///
+    /// Implements this API: <https://docs.boltprotocol.xyz/api/builder#get_header_with_proofs>
```

Comment:
update to https://docs.boltprotocol.xyz/technical-docs/api/builder#get_header_with_proofs